### PR TITLE
Upload photo

### LINKED ISSUE #64: Add createPhoto mutation
Add createPhoto mutation to upload a photo (of either a user or a dog) to AWS and save the URL to our database

### PR DESCRIPTION
**Pull request by chakeresa**
**Changes proposed in this pull request:**
 - Add `createPhoto` mutation, which uploads a photo to AWS and saves its URL to the database (if the request is for the current user or the current user's dog and the file is a valid image file.
 - Add `PhotoTypeInput` 
 - Add `BaseMutation#boot_unauthorized_user`
 - Add `file` to the GraphQL context (from the `file` query parameter)
 - Add a `caption` attribute to the `photos` table and `PhotoType`
 - Add test fixtures of a real photo and an audio file
 - Turns VCR off for requests to AWS (because the filename is going to change for every request)

Resolves #64, related to user stories #57 & #59 

**The following checks have been completed:**
 - [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
 - [x] Checked `coverage/index.html` - did not add any new code that's not covered by testing (if possible)
 - [x] Merged in the latest master to my branch with `git pull origin master` & resolved merge conflicts
 - [x] Ran `rails db:migrate`
 - [x] Ran the test suite - all tests are passing (or maybe skipped)
 - [x] Checked affected endpoints in Postman / GraphiQL
 - [x] Updated README for changes (new endpoints, new gems, etc)

**Notes:**
 - Includes a DB migration -- you'll have to run it on your machine

### CODE REVIEW

File: spec/graphql/mutations/create_photo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      ) {
+        photo {
+          #{photo_type_attributes}
+        }
+      }
+    }"
+  end
+end
+
+VCR.turn_on!
```
Comment (chakeresa) [inline spec/graphql/mutations/create_photo_spec.rb]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -356,6 +362,48 @@ Example of expected response:
 }
 ```
 
+#### createPhoto(photo: <PhotoInputType>)
+
+Uploads the photo from the `file` query param to an AWS S3 buckets and creates a photo resource in the database for the current user or the current user's dog. Whitelisted image file types include: bmp, jpeg, jpg, tiff, png. Requires a PhotoInputType argument. Returns a PhotoType object.
```
Comment (alexander-mathieu) [inline README.md]:
```suggestion
Uploads the photo from the `file` query param to an AWS S3 bucket and creates a photo resource in the database for the current user or the current user's dog. Whitelisted image file types include: bmp, jpeg, jpg, tiff, png. Requires a PhotoInputType argument. Returns a PhotoType object.
```

File: app/controllers/graphql_controller.rb [hunk 3]
```diff
@@ -4,7 +4,8 @@ def execute
     query = params[:query]
```
Comment (alexander-mathieu) [inline app/controllers/graphql_controller.rb]: 👍

File: app/graphql/mutations/base_mutation.rb [hunk 4]
```diff
@@ -7,5 +7,9 @@ def boot_unauthenticated_user
         raise GraphQL::ExecutionError, 'Unauthorized - a valid token query parameter is required'
       end
     end
+
```
Comment (alexander-mathieu) [inline app/graphql/mutations/base_mutation.rb]: I need this!

**Pull request merged**